app/api/init_api.py

```diff
@@ -162,6 +171,10 @@ async def on_shutdown() -> None:
         await app.state.services.database.disconnect()
         await app.state.services.redis.close()
 
+        if app.state.services.amqp is not None:
+            await app.state.services.amqp_channel.close()
+            await app.state.services.amqp.close()
```

Comment:
should this be `if app.settings.RABBITMQ_ENABLED`?

Reply:
Both would work but generally I would say this condition makes more sense, checking if the object you close is not null instead of checking if the object is technically initialized in the first place.